esmvalcore/preprocessor/_io.py

```diff
@@ -28,6 +28,41 @@
 }
 
 
+def _fix_aux_factories(cube):
+    """Fix :class:`iris.aux_factory.AuxCoordFactory` after concatenation.
+
+    Necessary because of bug in :mod:`iris` (see issue #2478).
+
+    """
+    coord_names = [coord.name() for coord in cube.coords()]
+
+    # Hybrid sigma pressure coordinate
+    if 'atmosphere_hybrid_sigma_pressure_coordinate' in coord_names:
```

Comment:
This looks like it could only work for atmospheric models? @ledm Any opinion?

Reply:
This can be easily extended to arbitrary [aux factories](https://scitools.org.uk/iris/docs/latest/iris/iris/aux_factory.html), but I don't have any data to test that/don't know if there are any issues which I'm not yet aware of. So I would suggest to stick with those variables in this PR.